app/views/template_invocations/show.js.erb

```diff
@@ -0,0 +1,21 @@
+var scrolledToBottom = ($(window).scrollTop() + $(window).height() == $(document).height());
+
+var lines = '<%=j render :partial => 'output_line', :collection => @lines, :locals => { :line_counter => params[:line_counter].to_i } %>';
+var target = $("div.terminal div.printable");
+
+<% if params[:since] %>
+  target.append(lines);
+<% else %>
+  target.html(lines);
+<% end %>
+
+<% if @template_invocation_task.pending? %>
+  $(function() {
+    delayed_refresh($("div.terminal").data('refresh-url'),
+            {"since": "<%= @lines.last['timestamp'] %>", "line_counter": $("div.terminal div.line").size()});
```

Comment:
`@lines` can be empty, when there were no new lines since last time: it should fallback to previous `since` value